### DISCUSSION TITLE
Squiggly error lines are not showing when I'm using Flutter in VSCode

### DISCUSSION THREAD

**Discussion by bigdawg817** [microsoft/vscode-cpptools#12129] [Q&A]
Hi everybody,

Recently started a Flutter course and I'm using VSCode to write. For some reason the squiggly error lines aren't showing. I've followed every written guide I can find, watched a bunch of YT videos and read through forums but can only find solutions for C/C++, JS and Python, not Flutter/Dart. 

I've trawled through the endless personalization settings within the console and still no luck.

Does anyone have any suggestions or solutions for this? It's probably something very basic I'm missing but I have spent hours trying to figure this out before asking for help. Appreciate all responses, thanks.

**Comment by bobbrow** [comment]
Did you open a folder, or just individual files?

By default we don't show the error squiggles if any of the `#include` lines in your source file fail to resolve.  It usually means a configuration is incomplete and we don't want to drown you with errors.  You can override this behavior with the `C_Cpp.errorSquiggles` setting.

**Reply by bobbrow** [reply]
If you can share more information about your setup, we can probably help you get it working.  Please also run the `C/C++: Log Diagnostics` command while the source file without the squiggles is focused in the editor and share the output.  That lets us know what's going on internally.

**Comment by bigdawg817** [comment]
Thanks for your reply.
I've tried updating VSCode, restored default settings, the C_Cpp.errorSquiggles setting isn't resolving it. I don't have an option for C/C++: Log Diagnostics showing from the command, I've attached a word doc with screenshots. 

Sorry I'm new to VSCode and Flutter, I know this is probably a really simple setting which I'm missing but I've exhausted everything and not sure what I can do to fix this

[Untitled document (1).docx](https://github.com/microsoft/vscode-cpptools/files/14717726/Untitled.document.1.docx)

**Reply by bobbrow** [reply]
In your screenshot I see a tab with a file called `main.dart`.  Do the files in your flutter/dart application have .cpp or .h extensions?  I don't know much about Flutter, but the example I saw online for main.dart uses a language that doesn't look like C++.  The "C/C++: Log Diagnostics" command is only available when you're looking at a C++ file.

**Reply by bobbrow** [reply]
If you're just using their language and not C++, I think you need to install the Flutter/Dart extensions:  https://docs.flutter.dev/tools/vs-code
